mockbroker.go

```diff
@@ -168,6 +180,23 @@ func NewMockBrokerAddr(t TestState, brokerID int32, addr string) *MockBroker {
 	return broker
 }
 
-func (b *MockBroker) Returns(e encoder) {
-	b.expectations <- e
+func (b *MockBroker) Returns(response encoder) *MockBrokerExpectation {
+	expectation := &MockBrokerExpectation{response: response}
+	b.expectations <- expectation
+	return expectation
```

Comment:
This now returns itself so the methods below can be chained.

Technically this is a bit shitty, because we immediately add it to the expectations channel, but allow it to be altered later. I guess this could cause concurrency issues, but because of the synchronous way our tests are set up I don't think this will cause issues.